### ISSUE TITLE
New 5 chars PDB IDs

### ISSUE THREAD

**Issue by joaomcteixeira** [julie-forman-kay-lab/IDPConformerGenerator#250]
This affects us:

https://www.wwpdb.org/news/news?year=2023#63ff72ccc031758bf1c30ff7

**Comment by menoliu** [comment]
I thought the CCD (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4393513/#:~:text=Summary%3A%20The%20Chemical%20Component%20Dictionary,Data%20Bank%20(PDB)%20entries.) was different from PDB ID? However I read that PDB IDs will still be affected in the future: https://www.wwpdb.org/news/news?year=2021#607760112786e73a79c76f9d

Are we thinking of making updates to `libpdb`? We would need an example of a culled list of 5 char PDB IDs to test out `pdbdl` and our `Structure` class. Thoughts?

**Comment by menoliu** [comment]
```
PRD_999999 1 1  THR THR N Y 
PRD_999999 1 2  X2AVD VAL N Y 
PRD_999999 1 3  PRO PRO N Y 
PRD_999999 1 4  SAR GLY N Y 
PRD_999999 1 5  MVA VAL N Y 
PRD_999999 1 6  PXZ ?   N Y 
PRD_999999 1 7  THR THR N Y 
PRD_999999 1 8  X2AVD VAL N Y 
PRD_999999 1 9  PRO PRO N Y 
PRD_999999 1 10 SAR GLY N Y 
PRD_999999 1 11 MVA VAL N Y 
```

Aha, I see in this example of a 5 character CCD of what was `DVA` now it's `X2AVD`. I was looking at our `libstructure` and `libcif` on line 333 and currently we organize our columns for mmCIF using `_atom_site.XYZ` (for models in wwPDB/extended-wwPDB-identifier-examples/tree/main) so it should still be okay?

**Comment by menoliu** [comment]
Yes I did some testing and we still recognise these residues and process these new mmCIF files no problem, just having trouble converting them to their one letter code constituents... but then again, I was thinking for phosphorylation PTM at least we'll have the lower-case sequence (e.g. phospho-serine could be `s` instead of `S`). Since phosphorylation is a very common PTM and I need to have it as a 1 letter code to perform sequence electrostatic potential analysis (in my new functions 😉)

**Comment by formankay** [comment]
Good to hear. Good luck coming up with 1-letter PTM codes. Happy to brainstorm if you want....
Thanks.
Julie

________________________________

**Comment by joaomcteixeira** [comment]
> Yes I did some testing and we still recognise these residues and process these new mmCIF files no problem,

Yes, the [parser](https://github.com/julie-forman-kay-lab/IDPConformerGenerator/blob/2a0e8747880aab815d74f7714d8cef15f0eafe82/src/idpconfgen/libs/libcif.py#L371) looks for spaces.

For the new `pdb_00001abc` codes we likely need to update the code below so that IDPCG recognises these new codes in the culled list or in general. 

https://github.com/julie-forman-kay-lab/IDPConformerGenerator/blob/2a0e8747880aab815d74f7714d8cef15f0eafe82/src/idpconfgen/libs/libpdb.py#L248-L252

Another point of possible error is the DSSP calculation. I don't remember how the PDBIDs are handled there.

To-do list:

- [ ] review PDBList and associated classes in `libpdb`
- [ ] review how PDBIDs are handled in DSSP (internally with IDPCG)
- [ ] the `aa3to1` can be extended if needed.
